src/DuneFormatter.ml

```diff
@@ -13,14 +14,30 @@ let format (document : TextDocument.t) _options _token =
   (* text of entire document *)
   let documentText = TextDocument.getText document fullDocumentRange in
 
-  Cmd.make ~cmd:(Path.ofString "dune") ()
-  |> Promise.Result.bind (fun duneCmd ->
-         Cmd.output duneCmd
-           ~args:[| "format-dune-file"  |]
-           ~stdin:documentText)
+  let cmd, args = Toolchain.getDuneFormatter toolchain in
+  Cmd.make ~cmd ()
+  |> Promise.then_ (function
```

Comment:
Can we use `Cmd.make ~cmd () >>= function` would read a bit cleaner. In general, we prefer the monadic API when possible